### PR TITLE
docs: Correct link targets and line highlights

### PR DESCRIPTION
**Pull request by michaelbrewer**
**Issue #, if available:**

## Description of changes:

Changes:
- Use target _blank for external links
- Correct some of the line highlighted in the examples
- Fix links to Tracer

**Checklist**

 

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] Update tests
* [x] Update docs
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/376ec0a2ac0d2a40e0af5717bef42ff84ca0d1b9/.github/semantic.yml#L2)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/utilities/middleware_factory.md [hunk 1]
```diff
@@ -78,9 +78,9 @@ This makes use of an existing Tracer instance that you may have initialized anyw
         ...
     ```
 
-When executed, your middleware name will [appear in AWS X-Ray Trace details as](../core/tracer) `## middleware_name`.
+When executed, your middleware name will [appear in AWS X-Ray Trace details as](/core/tracer) `## middleware_name`.
 
-For advanced use cases, you can instantiate [Tracer](../core/tracer) inside your middleware, and add annotations as well as metadata for additional operational insights.
+For advanced use cases, you can instantiate [Tracer](/core/tracer) inside your middleware, and add annotations as well as metadata for additional operational insights.
```
Comment (heitorlessa) [inline docs/utilities/middleware_factory.md:83]: Note to self: Investigate whether this broke due to Mike versioning

Relative path for internal linking should work 

https://www.mkdocs.org/user-guide/writing-your-docs/

Comment (heitorlessa) [reply docs/utilities/middleware_factory.md:83]: Just confirmed it works as expected, however the original is missing `.md` extension hence why it doesn't work.

`(..core/tracer.md)`

File: docs/utilities/validation.md [hunk 2]
```diff
@@ -20,7 +20,7 @@ You can validate inbound and outbound events using [`validator` decorator](#vali
 
 You can also use the standalone `validate` function, if you want more control over the validation process such as handling a validation error.
 
-We support any JSONSchema draft supported by [fastjsonschema](https://horejsek.github.io/python-fastjsonschema/) library.
+We support any JSONSchema draft supported by [fastjsonschema](https://horejsek.github.io/python-fastjsonschema/){target="_blank"} library.
```
Comment (heitorlessa) [inline docs/utilities/validation.md:23]: thank you :)

**Review by heitorlessa — CHANGES_REQUESTED** [review]
Two minor changes before I merge - Revert one change (relative is safer) and add `.md` suffix to a previously broken link

**Comment by michaelbrewer** [comment]
> Two minor changes before I merge - Revert one change (relative is safer) and add `.md` suffix to a previously broken link


Ok, but during local testing i get a 404?

**Comment by heitorlessa** [comment]
Working for me locally using both `make docs-local` and `make docs-local-docker`

What breaks is the second link because it doesn't have a `.md`, so MKdocs can't construct the correct internal link based on the official docs

image

image


then fixing with the correct `.md`:

```
When executed, your middleware name will [appear in AWS X-Ray Trace details as](../core/tracer.md) `## middleware_name`.

For advanced use cases, you can instantiate [Tracer](../core/tracer.md) inside your middleware, and add annotations as well as metadata for additional operational insights.
```

image

**Comment by michaelbrewer** [comment]
> Working for me locally using both `make docs-local` and `make docs-local-docker`
> 
> 
> 
> What breaks is the second link because it doesn't have a `.md`, so MKdocs can't construct the correct internal link based on the official docs
> 
> 
> 
> image
> 
> 
> 
> image
> 
> 
> 
> 
> 
> then fixing with the correct `.md`:
> 
> 
> 
> ```
> 
> When executed, your middleware name will [appear in AWS X-Ray Trace details as](../core/tracer.md) `## middleware_name`.
> 
> 
> 
> For advanced use cases, you can instantiate [Tracer](../core/tracer.md) inside your middleware, and add annotations as well as metadata for additional operational insights.
> 
> ```
> 
> 
> 
> image
> 
> 

Sweet. I will try again

**Comment by michaelbrewer** [comment]
@heitorlessa ok i updated it.

**Comment by michaelbrewer** [comment]
@heitorlessa  Note, we now have an issue for the link to the missing Event handler page:

```markdown
| [Event handler](./core/event_handler) | Event handler decorators for common Lambda events
```

**Pull request merged**